components/company/EditForm.vue

```diff
@@ -0,0 +1,370 @@
+  const form = ref()
+  const file = ref()
+
+  // Boolean control states
+  const editMode = computed(() => !!state.uid || !!state.companyUID)
+
+  // update ref on input change
+  const setLogo = (event) => {
+    file.value = event.target.files[0]
+  }
+
+  // update logo
+  const updateLogo = async () => {
+    if (!file.value || !state.uid) return
+
+    // creating multipart form data
+    const formData = new FormData()
+    formData.append('file', file.value)
+    formData.append('companyUID', state.uid)
+
+    // posting image to storage bucket
+    await $fetch('/api/company/logo', {
+      method: 'POST',
+      body: formData,
+    })
+      .then(async ({ URL }) => {
+        // catch url in response and update company logo
+        await $fetch('/api/company', {
+          method: 'PUT',
+          body: {
+            companyUID: state.uid,
+            logo: URL,
+          },
+        })
+          .then(() => {
+            displaySuccessAlert('alert.success.company.edit.modified')
+            setTimeout(() => navigateTo('/admin/companies'), 2000)
+          })
+          .catch(() =>
+            displayErrorAlert('alert.error.company.edit.modified_logo')
+          )
+      })
+      .catch(() => displayErrorAlert('alert.error.company.edit.modified'))
+  }
+
+  // remove logo from company
+  const deleteLogo = async () => {
+    await $fetch('/api/company', {
+      method: 'PUT',
+      body: {
+        logo: null,
+        companyUID: state.uid,
+      },
+    })
+      .then(() => {
+        displaySuccessAlert('alert.success.company.edit.deleted_logo')
+        setTimeout(() => navigateTo('/admin/companies'), 2000)
+      })
+      .catch(() => displayErrorAlert('alert.error.company.edit.deleted_logo'))
+  }
+
+  // save changes to existing company
+  const saveChanges = async () => {
+    // throw error on invalid form
+    const { valid } = await form.value.validate()
+    try {
+      if (!valid) throw new Error('Form is not valid')
+    } catch (error) {
+      displayErrorAlert('alert.error.form.invalid')
+      return
+    }
+
+    state.companyUID = state.uid
+
+    // update company
+    await $fetch('/api/company', {
+      method: 'PUT',
+      body: state,
+    })
+      .then(() => {
+        displaySuccessAlert('alert.success.company.edit.modified')
+        setTimeout(() => navigateTo('/admin/companies'), 2000)
+      })
+      .catch((error) => displayErrorAlertFromMessage(error.statusMessage))
+  }
+
+  const createEvent = async () => {
+    // throw error on invalid form
+    const { valid } = await form.value.validate()
+    try {
+      if (!valid) throw new Error('Form is not valid')
+    } catch (error) {
+      displayErrorAlert('alert.error.form.invalid')
+      return
+    }
+
+    state.companyUID = state.uid
+
+    // create company
+    await $fetch('/api/company', {
+      method: 'POST',
+      body: state,
+    })
+      .then(() => updateLogo()) // add logo newly created company if provided
```

Comment:
I don't know why, but this doesn't seem to run when creating a new company. I have to go and add the logo with the _update logo_ button after creating the company.